lib/src/modules/task.cc

```diff
@@ -0,0 +1,32 @@
+#include <pxp-agent/modules/task.hpp>
+
+#define LEATHERMAN_LOGGING_NAMESPACE "puppetlabs.pxp_agent.modules.task"
+#include <leatherman/logging/logging.hpp>
+
+namespace PXPAgent {
+namespace Modules {
+
+namespace fs = boost::filesystem;
+
+// TODO: construct and pass metadata to a new ExternalModule constructor, rather than calling `task metadata`
```

Comment:
I propose this, rather than overriding getModuleMetadata, because override functions aren't called during base class construction. So it didn't work.